### PR TITLE
Add support for python 3.6

### PR DESCRIPTION
**Pull request by joehybird**
- Fix tox & travis targets.
- Force elasticsearch server version to 1.5.2 by default
- Drop support for python 2.6
- Fix urllib3>=1.10 api break (raise NewConnectionError instead of ConnectionError)

### CODE REVIEW

File: pyelasticsearch/tests/client_tests.py [hunk 1]
```diff
@@ -1,7 +1,11 @@
 # coding=utf-8
 
-from mock import ANY, patch
-from nose.tools import eq_, ok_, assert_raises, assert_not_equal
+from six import PY3
+
+if PY3:
+    from unittest.mock import ANY, patch
```
Comment (erikrose) [inline pyelasticsearch/tests/client_tests.py]: Oddly, I don't even see `patch` used anywhere, in any file. I'm not sure why we were importing it.

File: tox.ini [hunk 2]
```diff
@@ -4,10 +4,25 @@
...
 [testenv]
-commands = python setup.py test
+setenv =
+    ESEARCH_VERSION=1.7.4
+    ESEARCH_PATH={toxworkdir}
+whitelist_externals =
+    sh
+deps =
+    certifi
+    elasticsearch>=1.3.0,<2.0.0
```
Comment (erikrose) [inline tox.ini]: We probably don't need to specify our deps here; `setup.py test`, below, will install them according to setup.py's `install_requires`. As a bonus, we don't have to repeat ourselves. :-)

File: pyelasticsearch/tests/client_tests.py [hunk 3]
```diff
@@ -273,8 +279,8 @@ def test_bulk(self):
                     index='test-index',
                     doc_type='book')
         except BulkError as exc:
-            eq_(exc.successes, [])
-            eq_(exc.errors, [{'index': {'status': 409,
+            self.assertEqual(exc.successes, [])
+            self.assertEqual(exc.errors, [{'index': {'status': 409,
                                         '_type': 'book',
```
Comment (erikrose) [inline pyelasticsearch/tests/client_tests.py]: We should fix the indentation in places like this.

File: tox.ini [hunk 4]
```diff
@@ -4,10 +4,25 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py26, py27, py33, py34, pypy
+envlist = py26, py27, py33, py34, py36, pypy
```
Comment (erikrose) [inline tox.ini]: We might as well remove python 2.6 here as well. And from the trove classifiers in setup.py.

**Comment by erikrose** [comment]
Thanks for fixing things up!

File: pyelasticsearch/tests/client_tests.py (hunk 1 above)
Comment (joehybird) [reply pyelasticsearch/tests/client_tests.py]: Here :)
https://github.com/joehybird/pyelasticsearch/blob/72a90ab76d28721b3c134c91cb1825f6ea3d1fae/pyelasticsearch/tests/client_tests.py#L90

File: tox.ini (hunk 2 above)
Comment (joehybird) [reply tox.ini]: Ok, but setup.py install doesn't work with tox.So I created a requirements.txt file that be used by both tox.ini and setup.py

File: tox.ini (hunk 4 above)
Comment (joehybird) [reply tox.ini]: Fixed

File: pyelasticsearch/tests/client_tests.py (hunk 3 above)
Comment (joehybird) [reply pyelasticsearch/tests/client_tests.py]: Fixed too.

File: pyelasticsearch/tests/client_tests.py (hunk 1 above)
Comment (erikrose) [reply pyelasticsearch/tests/client_tests.py]: Weird! Not sure how I missed it.

File: setup.py [hunk 5]
```diff
@@ -69,22 +76,16 @@ def find_version(file_path):
...
-        'certifi',
-        'elasticsearch>=1.3.0,<2.0.0',
-        'urllib3>=1.8,<2.0',
-        'simplejson>=3.0',
-        'six>=1.4.0,<2.0'
-    ],
-    tests_require=['mock', 'nose>=1.2.1'],
-    test_suite='nose.collector',
+    install_requires=install_requirements('requirements.txt'),
+    tests_require=['mock'] if PY2 else [],
```
Comment (erikrose) [inline setup.py]: We can get away with this as long as we don't release a wheel. (Those don't have the opportunity to run this conditional.) So we should be fine for now.

**Comment by erikrose** [comment]
Big thanks! So nice to have things modernized.

**Comment by joehybird** [comment]
I'm glad that you merged my work so fast ! I was a bit worried by the last commit date :)
And sorry for some response delays there's about 8 hours of time gap between us :)
It should be great to have an official release or I will make a temporarily one for my project.

**Comment by erikrose** [comment]
I just haven't been using ES lately, but pyelasticsearch continues to hum on happily in production. I'll put out a new release soon and let you know. Thanks again!

**Comment by erikrose** [comment]
Released! It's not showing up yet on https://pypi.python.org/pypi/pyelasticsearch/, but you can see it on https://pypi.org/project/pyelasticsearch/. Sorry for the delay; I was traveling. And thanks again for the good update!

**Pull request merged**